pkg/controller/manager/node/controller.go

```diff
@@ -62,6 +69,16 @@ func (h Handler) OnChange(key string, node *corev1.Node) (*corev1.Node, error) {
 	return node, nil
 }
 
+func (h Handler) OnRemove(key string, node *corev1.Node) (*corev1.Node, error) {
+	if node == nil {
+		return nil, nil
+	}
+
+	klog.Infof("node %s is removed", node.Name)
```

Comment:
validate https://github.com/harvester/harvester/issues/3697 failed
```bash
❯ k get no
NAME        STATUS   ROLES                       AGE   VERSION
harv-11-1   Ready    control-plane,etcd,master   45m   v1.24.11+rke2r1

❯ k logs -f harvester-network-controller-manager-68567bf59c-9j25k
I0323 09:06:49.036815       1 main.go:105] Starting network controller with 2 threads.
I0323 09:06:49.099493       1 main.go:110] Starting network controller in namespace: harvester-system.
W0323 09:06:49.100192       1 client_config.go:617] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0323 09:06:49.142814       1 leaderelection.go:248] attempting to acquire leader lease kube-system/harvester-network-controllers...
I0323 09:07:35.159675       1 leaderelection.go:258] successfully acquired lease kube-system/harvester-network-controllers
time="2023-03-23T09:07:56Z" level=info msg="Do not update the existing CRD NetworkAttachmentDefinition"
E0323 09:08:07.835543       1 memcache.go:206] couldn't get resource list for subresources.kubevirt.io/v1: the server is currently unable to handle the request
E0323 09:08:08.034011       1 memcache.go:206] couldn't get resource list for custom.metrics.k8s.io/v1beta1: the server is currently unable to handle the request
E0323 09:08:08.034715       1 memcache.go:206] couldn't get resource list for subresources.kubevirt.io/v1alpha3: the server is currently unable to handle the request
E0323 09:08:08.035874       1 memcache.go:206] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request
E0323 09:08:08.139580       1 memcache.go:104] couldn't get resource list for subresources.kubevirt.io/v1: the server is currently unable to handle the request
E0323 09:08:08.235380       1 memcache.go:104] couldn't get resource list for subresources.kubevirt.io/v1alpha3: the server is currently unable to handle the request
E0323 09:08:08.237618       1 memcache.go:104] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request
E0323 09:08:08.239994       1 memcache.go:104] couldn't get resource list for custom.metrics.k8s.io/v1beta1: the server is currently unable to handle the request
time="2023-03-23T09:08:08Z" level=info msg="Starting network.harvesterhci.io/v1beta1, Kind=ClusterNetwork controller"
time="2023-03-23T09:08:08Z" level=info msg="Starting k8s.cni.cncf.io/v1, Kind=NetworkAttachmentDefinition controller"
time="2023-03-23T09:08:08Z" level=info msg="Starting network.harvesterhci.io/v1beta1, Kind=LinkMonitor controller"
time="2023-03-23T09:08:08Z" level=info msg="Starting network.harvesterhci.io/v1beta1, Kind=VlanConfig controller"
time="2023-03-23T09:08:08Z" level=info msg="Starting /v1, Kind=Node controller"
I0323 09:08:08.338537       1 controller.go:51] vlan config test has been changed, spec: {Description: ClusterNetwork:test NodeSelector:map[] Uplink:{NICs:[ens34] LinkAttrs:0xc0011d1620 BondOptions:0xc001cb84e0}}
time="2023-03-23T09:08:08Z" level=info msg="Starting network.harvesterhci.io/v1beta1, Kind=VlanStatus controller"
I0323 09:08:08.433245       1 controller.go:64] vlan status test-harv-11-1-eec5db85 has been changed, node: harv-11-1, clusterNetwork: test, vc: test
time="2023-03-23T09:08:08Z" level=info msg="Starting batch/v1, Kind=Job controller"
I0323 09:14:49.734994       1 controller.go:51] vlan config test has been changed, spec: {Description: ClusterNetwork:test NodeSelector:map[] Uplink:{NICs:[ens34] LinkAttrs:0xc00108ec90 BondOptions:0xc0007ebcf8}}
I0323 09:16:40.758308       1 controller.go:64] vlan status test-harv-11-2-77cc8a3f has been changed, node: harv-11-2, clusterNetwork: test, vc: test
I0323 09:16:40.776444       1 controller.go:64] vlan status test-harv-11-2-77cc8a3f has been changed, node: harv-11-2, clusterNetwork: test, vc: test
I0323 09:16:42.269319       1 controller.go:64] vlan status test-harv-11-2-77cc8a3f has been changed, node: harv-11-2, clusterNetwork: test, vc: test
I0323 09:22:02.362875       1 controller.go:77] node harv-11-2 is removed
I0323 09:22:02.434077       1 controller.go:77] node harv-11-2 is removed
I0323 09:22:02.463832       1 controller.go:77] node harv-11-2 is removed

❯ kubectl get lms -o yaml
apiVersion: v1
items:
- apiVersion: network.harvesterhci.io/v1beta1
  kind: LinkMonitor
  metadata:
    creationTimestamp: "2023-03-23T08:49:59Z"
    finalizers:
    - wrangler.cattle.io/harvester-network-link-monitor-controller
    generation: 8
    name: mgmt
    resourceVersion: "38159"
    uid: be0c796c-ea18-4c9b-b682-2fce3b443c5e
  spec:
    targetLinkRule:
      nameRule: mgmt(-br|-bo)
  status:
    linkStatus:
      harv-11-1:
      - index: 4
        mac: "50:50:50:50:50:37"
        name: mgmt-br
        state: up
        type: bridge
      - index: 5
        mac: "50:50:50:50:50:37"
        masterIndex: 4
        name: mgmt-bo
        state: up
        type: bond
- apiVersion: network.harvesterhci.io/v1beta1
  kind: LinkMonitor
  metadata:
    creationTimestamp: "2023-03-23T08:49:58Z"
    finalizers:
    - wrangler.cattle.io/harvester-network-link-monitor-controller
    generation: 25
    name: nic
    resourceVersion: "38200"
    uid: 93eb506a-8681-4ca2-a2ee-9022570b7787
  spec:
    targetLinkRule:
      typeRule: device
  status:
    linkStatus:
      harv-11-1:
      - index: 2
        mac: "50:50:50:50:50:37"
        masterIndex: 5
        name: ens33
        state: up
        type: device
      - index: 3
        mac: 00:0c:29:38:4f:5e
        masterIndex: 73
        name: ens34
        state: up
        type: device
      harv-11-2:
      - index: 2
        mac: "50:50:50:50:50:38"
        masterIndex: 5
        name: ens33
        state: up
        type: device
      - index: 3
        mac: 00:0c:29:94:50:12
        masterIndex: 25
        name: ens34
        state: up
        type: device
- apiVersion: network.harvesterhci.io/v1beta1
  kind: LinkMonitor
  metadata:
    creationTimestamp: "2023-03-23T08:56:54Z"
    finalizers:
    - wrangler.cattle.io/harvester-network-link-monitor-controller
    generation: 28
    name: test
    resourceVersion: "38164"
    uid: 87c3e38a-eb22-4945-a525-c7e4f5a96a62
  spec:
    targetLinkRule:
      nameRule: test(-br|-bo)
  status:
    linkStatus:
      harv-11-1:
      - index: 13
        mac: 4e:d7:ce:03:46:fa
        name: test-br
        promiscuous: true
        state: up
        type: bridge
      - index: 73
        mac: 00:0c:29:38:4f:5e
        masterIndex: 13
        name: test-bo
        state: up
        type: bond
      harv-11-2:
      - index: 24
        mac: e6:13:30:9f:66:e5
        name: test-br
        promiscuous: true
        state: down
        type: bridge
      - index: 25
        mac: 00:0c:29:94:50:12
        masterIndex: 24
        name: test-bo
        state: up
        type: bond
kind: List
metadata:
  resourceVersion: ""
```

Reply:
This is an occasional problem. When a node is deleted, if the corresponding harvester network controller pod has not been deleted or failed, link statuses will be added again.
I fixed it by ignoring the removed node when the harvester network controller pod adds link statuses.

Reply:
Thanks! Now the test has passed